### PR TITLE
Port library from `BigDecimal` to `Double`, and add QoL improvements

### LINKED ISSUE #23: Memory consumption issue
Hi @ggalmazor!
I submitted a PR/opened an issue with a bug in your library about a year ago. Thanks again for your library; it's been incredibly useful in my project. Now, I'm reaching out for a feature request I'd like to implement in your library 😊

## Context
In my project, we utilize the down-sampling library to visualize large datasets (approximately 1,000,000 points) on a web interface that can handle only a few thousand points. These points are timeseries with two attributes: timestamp (`Date`) and measure (`long`). To integrate with your library, I need to extend this class with `com.ggalmazor.ltdownsampling.Point`, which uses two `BigDecimal`. These two additions cost at least [32b](https://stackoverflow.com/questions/2501176/java-bigdecimal-memory-usage) x 2 per point.
About a month ago, we encountered a memory issue when multiple requests were made simultaneously. As a quick fix, we increased the application's memory allocation. However, we're now seeking a long-term solution that would be more memory-efficient.

## Proposed Implementation
I'm considering creating a new class, lighter than `Point`, without `BigDecimal` attributes `x` and `y`. Instead, it would have only getters and setters performing on-the-fly conversion between the attributes and the `BigDecimal` value used by the algorithm. This implementation would introduce a new class without altering the core of the library. The `Point` class would still be available for backward compatibility.

## Potential Blockers
Are there any potential blockers for developing this implementation? I'm thinking about:
- Utilizing specific `BigDecimal` methods
- Intensively using `BigDecimal` values, which might negatively impact algorithm performance when converting values on-the-fly multiple times
- Any other considerations?

I'd be thrilled to update the library in this direction and open a PR for these changes. Do you agree with updating the library in this direction?

P.S.: Is there a chance of duplicated arrays during down-sampling (resulting in extra memory usage)?

### PR DESCRIPTION
**Pull request by ggalmazor**
Fixes #23

Problem to be solved:
- `BigDecimal` is not an optimal solution for large data series, causing excessive memory consumption. Better alternatives can be used without losing precision.
- The library is too opinionated regarding the input types that must be provided. Users need to either use or extend `Point`, which is a concrete class, and they are bound to its internal types.
- The current implementation has some redundant bits that can be trimmed off

In this PR:
- Port `Point` from `BigDecimal` to `Double`
- Port the `Point` factory to accept `Number` objects for maximum compatibility
- Move the `Point` class to `DoublePoint` and extract its interface into `Point`
- Move point operations on `DoublePoint` objects (`add`, `subtract`, `half`) to `Point` and make static
- Migrate tests to a reusable custom matcher that compares actual `x` and `y` values instead of relying on equality.
- Remove the custom `equals` and `hashCode` everywhere. This was only needed for the tests.
- Random minor cleanup

### CODE REVIEW

File: src/main/java/com/ggalmazor/ltdownsampling/Point.java [hunk 1]
```diff
@@ -6,50 +6,50 @@
 
 public class Point {
   private static final MathContext MC = MathContext.UNLIMITED;
```
Comment (capsuleman) [inline src/main/java/com/ggalmazor/ltdownsampling/Point.java]: Is MathContext still have a use here?  
There is useless import remaining

**Review by capsuleman — APPROVED** [review]
Great work! 🤩

**Comment by capsuleman** [comment]
It looks good for me to merge your PR, is it ok for you?

**Comment by ggalmazor** [comment]
Sorry, @capsuleman , I've been super busy with work this week. Let's merge it a roll out a `0.1.0` version

**Comment by ggalmazor** [comment]
Alright, it's done!

The package is available now at https://jitpack.io/#ggalmazor/lt_downsampling_java8/0.1.0

**Comment by capsuleman** [comment]
No problem! Thank you again for your help, you rock! ⭐

**Pull request merged**